airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -378,7 +379,7 @@ def create_new_pod_for_operator(self, labels, launcher) -> Tuple[State, k8s.V1Po
         )
 
         self.pod = pod
-        self.log.info("Starting pod %s", pod)
+        self.log.info("Starting pod:\n%s", yaml.safe_dump(pod.to_dict()))
```

Comment:
Are there any security implications of doing this? Exposing secrets etc in the logs

Reply:
By design, only metadata and a declarative configuration description should be included here. Secrets should be in K8s Secrets. 
See: https://kubernetes.io/docs/concepts/configuration/secret/

Reply:
I would still increase the log level as DEBUG i.e. you would only want the entire pod spec when you are debugging something unless there is a use-case which I have not thought off

Reply:
Good idea. I updated PR.